### PR TITLE
EIP-225 Clique Spec Improvements

### PR DESCRIPTION
**Pull request by holgerd77**
We have now finished our `TypeScript` Clique implementation (https://github.com/ethereumjs/ethereumjs-monorepo/pull/1032 as well as https://github.com/ethereumjs/ethereumjs-monorepo/pull/1074 and https://github.com/ethereumjs/ethereumjs-monorepo/pull/1088).

Thanks for this wonderful spec, the clarity of the writing and the extensive test cases helped us a lot! 😄  We nevertheless stumbled upon a few things which caused us some significant extra debugging time. Just took the time to extract, think this should be useful for future implementations.

//cc @karalabe

### CODE REVIEW

File: EIPS/eip-225.md [hunk 1]
```diff
@@ -110,27 +111,28 @@ We also define the following per-block constants:
...
 We repurpose the `ethash` header fields as follows:
 
- * **`beneficiary`**: Address to propose modifying the list of authorized signers with.
+ * **`beneficiary`** / **`miner`**: Address to propose modifying the list of authorized signers with.
    * Should be filled with zeroes normally, modified only while voting.
    * Arbitrary values are permitted nonetheless (even meaningless ones such as voting out non signers) to avoid extra complexity in implementations around voting mechanics.
    * **Must** be filled with zeroes on checkpoint (i.e. epoch transition) blocks.
-   * Transaction execution **must** use the actual block signer (see `extraData`) for the `COINBASE` opcode.
+   * Transaction execution **must** use the actual block signer (see `extraData`) for the `COINBASE` opcode and transaction fees **must** be attributed to the signer account.
+   * Block execution **must** not distribute any block rewards.
```
Comment (shemnon) [inline EIPS/eip-225.md:125]: This feels like an addition to the spec rather than a clarification like the other proposed edits are.  Enterprise chains may want to add block rewards to encourage miners to stay online.  

My opinion is that this should be removed from the edits or be come a **should** instead of a **must**

Comment (holgerd77) [reply EIPS/eip-225.md]: Thanks for having a look, definitely not my intention to change the spec in any way, have updated.

File: EIPS/eip-225.md [hunk 2]
```diff
@@ -110,27 +111,28 @@ We also define the following per-block constants:
...
 We repurpose the `ethash` header fields as follows:
 
- * **`beneficiary`**: Address to propose modifying the list of authorized signers with.
+ * **`beneficiary`** / **`miner`**: Address to propose modifying the list of authorized signers with.
    * Should be filled with zeroes normally, modified only while voting.
    * Arbitrary values are permitted nonetheless (even meaningless ones such as voting out non signers) to avoid extra complexity in implementations around voting mechanics.
    * **Must** be filled with zeroes on checkpoint (i.e. epoch transition) blocks.
-   * Transaction execution **must** use the actual block signer (see `extraData`) for the `COINBASE` opcode.
+   * Transaction execution **must** use the actual block signer (see `extraData`) for the `COINBASE` opcode and transaction fees **must** be attributed to the signer account.
+   * Block execution **should** not distribute any block rewards.
```
Comment (MicahZoltu) [inline EIPS/eip-225.md:125]:
```suggestion
```
This feels like either a normative change, or an unnecessary change and should be removed either way.  `**should**` recommendations usually don't provide value for standardization because you cannot rely on them.  If they are relied on in some way, then this feels like a normative change which cannot be made to final EIPs.

File: EIPS/eip-225.md [hunk 3]
```diff
@@ -141,9 +143,9 @@ We repurpose the `ethash` header fields as follows:
...
+To authorize a block for the network, the signer needs to sign the block's sighash containing **everything except the signature itself**. This means that this hash contains every field of the header (`nonce` and `mixDigest` included), and also the  `extraData` with the exception of the 65 byte signature suffix. The fields are hashed in the order of their definition in the yellow paper. Note that this sighash differs from the final block hash which also includes the signature.
 
-This hash is signed using the standard `secp256k1` curve, and the resulting 65 byte signature (`R`, `S`, `V`, where `V` is `0` or `1`) is embedded into the `extraData` as the trailing 65 byte suffix.
+The  sighash is signed using the standard `secp256k1` curve, and the resulting 65 byte signature (`R`, `S`, `V`, where `V` is `0` or `1`) is embedded into the `extraData` as the trailing 65 byte suffix.
```
Comment (MicahZoltu) [inline EIPS/eip-225.md:148]:
```suggestion
The sighash is signed using the standard `secp256k1` curve, and the resulting 65 byte signature (`R`, `S`, `V`, where `V` is `0` or `1`) is embedded into the `extraData` as the trailing 65 byte suffix.
```

**Review by MicahZoltu — CHANGES_REQUESTED** [review]

File: EIPS/eip-225.md (hunk 2 above)
Comment (holgerd77) [reply EIPS/eip-225.md:125]: Ok, just added this to help on implementation, but I can follow here along.

**Comment by MicahZoltu** [comment]
@karalabe Can you review this and make sure there are no normative changes (just clarification)?  If so, please do a GitHub review/approve and I'll merge.

**Comment by lightclient** [comment]
Bumping again - @karalabe if you could take a look please.

**Comment by alita-moore** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):


	 - eip-225.md is in state final, not draft or last call or review
	 - eip-225.md requires approval from one of (karalabe)
 @karalabe

**Comment by holgerd77** [comment]
Any chance that we get this merged?

Sorry Peter, I know about these nasty side tasks which nobody asked for and which sneak in from the sideline 😋.

I am also very willing to remove all things which have even the slightest controversial aspect. 😀

**Comment by MicahZoltu** [comment]
Someone may have to reach out to Peter via alternative means, since it doesn't appear that they are receiving pings here.

**Comment by karalabe** [comment]
Yeah, I just figured you'll auto-merge bike shedding fixes :P

**Review by karalabe — APPROVED** [review]

**Comment by MicahZoltu** [comment]
In this case, I don't know enough about the subject matter to be able to assert whether this is purely a non-normative or not, hence the need for author approval.

**Comment by holgerd77** [comment]
@karalabe we actually lost ~3 development days with debugging on these "trivialities", so bear in mind the non-enlightened. 😄 

Thanks for merging.

**Comment by holgerd77** [comment]
(e.g. we were sorting Clique signers on a FIFO basis and were searching our butts of why the block hashes are not matching 😄)
